### PR TITLE
Add  new recipe for pecat

### PR DESCRIPTION
**Pull request by lemene**
Add a new recipe for pecat

----

Please read the [guidelines for Bioconda recipes](https://bioconda.github.io/contributor/guidelines.html) before opening a pull request (PR).

* If this PR adds or updates a recipe, use "Add" or "Update" appropriately as the first word in its title.
* New recipes not directly relevant to the biological sciences need to be submitted to the [conda-forge channel](https://conda-forge.org/docs/) instead of Bioconda.
* PRs require reviews prior to being merged. Once your PR is passing tests and ready to be merged, please issue the `@BiocondaBot please add label` command.
* Please post questions [on Gitter](https://gitter.im/bioconda/Lobby) or ping `@bioconda/core` in a comment.

<details>
  <summary>Please use the following BiocondaBot commands:</summary>

Everyone has access to the following BiocondaBot commands, which can be given in a comment:

<table>
  <tr>
    <td><code>@BiocondaBot please update</code></td>
    <td>Merge the master branch into a PR.</td>
  </tr>
  <tr>
    <td><code>@BiocondaBot please add label</code></td>
    <td>Add the <code>please review & merge</code> label.</td>
  </tr>
  <tr>
    <td><code>@BiocondaBot please fetch artifacts</code></td>
    <td>Post links to CI-built packages/containers. <br />You can use this to test packages locally.</td>
  </tr>
</table>

For members of the Bioconda project, the following command is also available:

<table>
  <tr>
    <td><code>@BiocondaBot please merge</code></td>
    <td>Upload built packages/containers and merge a PR. <br />Someone must approve a PR first! <br />This reduces CI build time by reusing built artifacts.</td>
  </tr>
</table>

Also, the bot watches for comments from non-members that include `@bioconda/<team>` and will automatically re-post them to notify the addressed `<team>`.

</details>

### CODE REVIEW

**Comment by lemene** [comment]
@BiocondaBot please add label

File: recipes/pecat/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+
+about:
+  home: https://github.com/lemene/PECAT
+  license: BSD-2-Clause license
+  summary: A phased error correction and assembly tool
+
+extra:
+  skip-lints:
+    - missing_hash
+    - uses_vcs_url
```
Comment (johnlees) [inline recipes/pecat/meta.yaml:43]: Is it possible to use conda versions of any of the submodule dependencies instead?

File: recipes/pecat/build.sh [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+CXX=${CXX}
+
+mkdir -p ${PREFIX}/bin
+ln -fs $CC ${PREFIX}/bin/gcc
+ln -fs $CXX ${PREFIX}/bin/g++
+
+# modify the bugs in the makefiles
+echo "modify the bugs in the makefiles"
+sed -i 's/\$$/ -lrt /' $SRC_DIR/src/makefile
+sed -i 's/= ..\/build/= .\/build/' $SRC_DIR/makefile
```
[the remark this answers is not in the document]
Comment (lemene) [reply recipes/pecat/build.sh:30]: Thank you very much for your suggestion. I will change the modification to a patch.

File: recipes/pecat/meta.yaml (hunk 1 above)
Comment (bgruening) [reply recipes/pecat/meta.yaml:43]: It is recommended to not vendor other packages. Your thirdparty packages should be a dependency under the host section.

File: recipes/pecat/meta.yaml [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+
+requirements:
+  build:
+    - make
+    - {{ compiler('c') }}
+    - {{ compiler('cxx') }}
+  host:
+    - zlib
+  run:
+    - zlib
```
Comment (bgruening) [inline recipes/pecat/meta.yaml:25]: zlib is not needed in run

File: recipes/pecat/meta.yaml [hunk 4]
```diff
@@ -0,0 +1,43 @@
+{% set name = "pecat" %}
+{% set version = "0.0.1" %}
+
+package:
+  name: {{ name }}
+  version: {{ version }}
+
+
+source:
+  git_url: https://github.com/lemene/PECAT.git
```
Comment (bgruening) [inline recipes/pecat/meta.yaml:10]: please use a source tarball if possible

File: recipes/pecat/build.sh [hunk 5]
```diff
@@ -0,0 +1,49 @@
...
+export CPP_INCLUDE_PATH=${PREFIX}/include
+export CXX_INCLUDE_PATH=${PREFIX}/include
+export PATH=$PATH:${PREFIX}/bin
+
+
+echo "run git submodule update --init --recursive"
+git submodule update --init --recursive
+
+
+echo ${CC}
```
Comment (bgruening) [inline recipes/pecat/build.sh:17]: please clean that up a bit

File: recipes/pecat/meta.yaml (hunk 1 above)
Comment (lemene) [reply recipes/pecat/meta.yaml:43]: OK, I see. Thank you for your suggestion. I have another question. Should I close the pull request, modify the code and then open a new pull request?

File: recipes/pecat/meta.yaml (hunk 3 above)
Comment (lemene) [reply recipes/pecat/meta.yaml:25]: I get it. Thank you.

File: recipes/pecat/meta.yaml (hunk 4 above)
Comment (lemene) [reply recipes/pecat/meta.yaml:10]: I get it. Thank you.

File: recipes/pecat/build.sh (hunk 5 above)
Comment (lemene) [reply recipes/pecat/build.sh:17]: OK, thank you.

File: recipes/pecat/meta.yaml (hunk 1 above)
Comment (bgruening) [reply recipes/pecat/meta.yaml:43]: You can change it in place here, no need for a new one.

**Comment by lemene** [comment]
@BiocondaBot please add label

**Review by daler — APPROVED** [review]
all the changes look good, thanks

**Pull request merged**